### ISSUE TITLE
Improve error message: "Field set to nothing can't be serialized"

### ISSUE THREAD

**Issue by msfcolombo** [microsoft/bond#1017]
Customers have huge and complex pipelines. Bond serialization is buried deep in the code, and sporadic errors like a null value in some record may require an humongous amount of time such as connecting to the cloud service for debugging (before getting clearance for doing so), or adding more logs elsewhere (which would be duplicated when `Bond` is used for logging), and deploying again.

Therefore, this error message is not being helpful at all. It should state at very minimum the field that is not satisfying the condition, and ideally the schema name as well.

**Comment by chwarr** [comment]
Makes sense. Can you submit a PR to improve the message? The line that needs to be changed [is 135 in SimpleBinary.cs](https://github.com/microsoft/bond/blob/8d0842280856f357664d6ea9d3d010497444e8ab/cs/src/core/protocols/SimpleBinary.cs#L135).

**Comment by himanshusainig** [comment]
is there a workaround to this. The bigger issue is boolean with false values are also being omitted.
I need all the fields to be present in the serialized json as this will be deserialized in another application based c++ where these fields are not marked as nullable.

**Comment by chwarr** [comment]
@himanshusainig, this issue is specifically about a better error message when a `= nothing` field is encountered that cannot be serialized. Can you please create a new issue about your problem? It will be useful to see the .bond schemas that each app. Make sure you've tested with the currently released versions of Bond too, in case the issue has been fixed already.
